[PPML] Migrate SplitAndEncrypt From E2E To BigDL

piaolaidelangman opened the pull request.

jenniew commented:
For csv, Split and Encrypt may use PPMLContext.read.csv(), then repartition and PPMLContext.write.csv() to save to CSV with encryption.
For other textfile, Split and Encrypt may use PPMLContext.textFile(), then repartition and PPMLContext.saveAsTextFile() with encryption to save. @qiuxin2012 , will we add this api?
That may be faster for huge file.

jason-dai commented:
> For csv, Split and Encrypt may use PPMLContext.read.csv(), then repartition and PPMLContext.write.csv() to save to CSV with encryption. For other textfile, Split and Encrypt may use PPMLContext.textFile(), then repartition and PPMLContext.saveAsTextFile() with encryption to save. @qiuxin2012 , will we add this api? That may be faster for huge file.

Yes, the plan is to call PPMLContext read/write in the CLI tools

piaolaidelangman commented:
> > For csv, Split and Encrypt may use PPMLContext.read.csv(), then repartition and PPMLContext.write.csv() to save to CSV with encryption. For other textfile, Split and Encrypt may use PPMLContext.textFile(), then repartition and PPMLContext.saveAsTextFile() with encryption to save. @qiuxin2012 , will we add this api? That may be faster for huge file.
> 
> Yes, the plan is to call PPMLContext read/write in the CLI tools

I will close this pull request.
